### PR TITLE
[llvm-cov] format cells in code coverage report with 0/0 branches/functions/lines differently

### PR DESCRIPTION
**Pull request by hanickadot**
Currently such cells in the report are formatted as red. Which screams "problem" but with more and more rangefied code, there can be whole files without any branch inside them. 

Before:
Screenshot 2023-12-18 at 11 09 07

After:
Screenshot 2023-12-18 at 11 08 29

Things to notice:
-  highlighted line which is result of cursor hover (now whole line is highlighted)
-  before there were some borders doubled (fixed)
-  cells with `- 0/0` coverage are no longer red

### CODE REVIEW

File: llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp
Comment (chapuni) [inline llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp:328]: We don't use braces on a single-statement. Could you follow?

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

**Comment by hanickadot** [comment]
I also needed to fix one test which was matching multiple rows in the table instead of one.

**Review by chapuni — APPROVED** [review]
Thanks. This is a good improvement for the UX!
I pre-approve with one nit.

I would like to wait for at least one more for the approval. @petrhosek , how is this?

**Review by petrhosek — APPROVED** [review]

File: llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp
Comment (hanickadot) [reply llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp:375]: yes, of course! force of habit :)

**Comment by hanickadot** [comment]
I'm sorry I didn't notice you already looked into it, and I added also sorting mechanism. Can you look at it too?

File: llvm/test/tools/llvm-cov/coverage_watermark.test
Comment (hanickadot) [inline llvm/test/tools/llvm-cov/coverage_watermark.test:43]: previously the red was actually the - (0/0), so I changed the threshold and make the test more precise

**Comment by chapuni** [comment]
@hanickadot I hope we could land the 1st idea (when I approved) in advance.
Could you rewind extra changes and create another PR?

I expect sorting will be a good idea. (I also wish filtering)
It will introduce JS and it should be another PR.

**Comment by hanickadot** [comment]
yes, already rewinded

File: llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp
Comment (chapuni) [reply llvm/tools/llvm-cov/SourceCoverageViewHTML.cpp:328]: What about this?

**Review by chapuni — APPROVED** [review]
Thanks. Don't forget squashing.

**Comment by chapuni** [comment]
Pushed manually, 18e1179208a1

**Pull request closed without merging**
